dsgrid/registry/project_registry_manager.py

```diff
@@ -372,20 +372,18 @@ def _register_all_in_one_dimensions(
         new_dimensions = []
         new_mappings = {}
         dim_type_to_ref = {x.dimension_type: x for x in model.dimensions.base_dimension_references}
-        for dimension_type in model.dimensions.all_in_one_supplemental_dimensions:
-            if dimension_type == DimensionType.TIME:
-                raise DSGInvalidParameter("Cannot have time in all_in_one_supplemental_dimensions")
-            if dimension_type in new_mappings:
-                raise DSGInvalidParameter(
-                    f"{dimension_type} is listed twice in all_in_one_supplemental_dimensions"
-                )
-
+        for dimension_type in (x for x in DimensionType if x != DimensionType.TIME):
             dim_ref = dim_type_to_ref[dimension_type]
             dim_config = self._dimension_mgr.get_by_id(dim_ref.dimension_id)
             dt_str = dimension_type.value
-            dt_plural = f"all_{dt_str}s"
-            dt_plural_dash = f"all-{dt_str}s"
-            dt_plural_formal = f"All {dt_str.title()}s"
+            if dt_str.endswith("y"):
+                dt_str = dt_str[:-1]
+                suffix = "ies"
+            else:
+                suffix = "s"
+            dt_plural = f"all_{dt_str}{suffix}"
```

Comment:
what if dt_str contains "space", "dash", or "underscore"?

Reply:
`dt_str` will be a `DimensionType` value, and so it shouldn't have a space or dash. I suppose a future dimension type could have an underscore. I don't think I follow your concern. Can you elaborate?

Reply:
Not super important, just wondering if we also need to do dt_str.replace() according to the "space", "dash", "underscore" convention in the name.

Reply:
The only issue would be the possibility of a future dimension value containing an underscore. I don't think we have any rules about underscores in dimension/dimension_mapping record value IDs or names. Right?